java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -737,8 +737,10 @@ class DatabaseImpl(
                 """.trimIndent(),
                 arrayOf(nowMillis.toString())
             ).map { it.getLong(0) to it.getString(1) }.toSet()
-            val storageKeyIds = storageKeyIdsPairs.map { it.first }
-            val storageKeys = storageKeyIdsPairs.map { it.second }
+            val storageKeyIds = storageKeyIdsPairs.map { it.first.toString() }.toTypedArray()
+            val storageKeys = storageKeyIdsPairs.map { it.second }.toTypedArray()
+            // List of question marks of the same length, to be used in queries.
+            val questionMarks = storageKeyIdsPairs.map { "?" }.joinToString()
```

Comment:
Might be worth pulling this out into its own method, if for no other reason than to be able to write some KDoc explaining what it's doing and why it's necessary.